### PR TITLE
Nullable Instrumentation.EntityFrameworkCore

### PR DESCRIPTION
**Pull request by Kielek**
Towards #894 .

## Changes

Nullable Instrumentation.EntityFrameworkCore.
Introduce `IncludeSharedSpanHelper` and `ActivityInstrumentationHelper` to include files into compilation only when needed.

For significant contributions please make sure you have completed the following items:

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~
* [x] Changes in public API reviewed

### CODE REVIEW

**Review by ejsmith — APPROVED** [review]
LGTM

File: src/Shared/ListenerHandler.cs [hunk 1]
```diff
@@ -26,9 +29,11 @@ internal abstract class ListenerHandler
     /// Initializes a new instance of the <see cref="ListenerHandler"/> class.
     /// </summary>
     /// <param name="sourceName">The name of the <see cref="ListenerHandler"/>.</param>
-    public ListenerHandler(string sourceName)
+    /// <param name="supportsNullActivity">Indicates whether the  <see cref="ListenerHandler"/> supports NULL <see cref="Activity"/>.</param>
+    protected ListenerHandler(string sourceName, bool supportsNullActivity)
```
Comment (utpilla) [inline src/Shared/ListenerHandler.cs:33]: Why is this new `bool` parameter being added? I don't think we need this.

Comment (Kielek) [reply src/Shared/ListenerHandler.cs:33]: Previously there to return true by `SupportsNullActivity` we need to override this value in child class. For now, we have to inject by constructor. It was more intuitive for me.

The property is used under DiagnosticSourceListener -> OnNext method

Comment (utpilla) [reply src/Shared/ListenerHandler.cs:33]: Any particular benefit to updating the shared file?

I think we could achieve this by simply using `#nullable enable` in the individual classes and annotating the code accordingly (by using `?` and `!`).

Comment (Kielek) [reply src/Shared/ListenerHandler.cs]: Changes related to SupportNullActivity reverted in 7a455f055c158cb374308e738ee5c06c554477a9

**Review by utpilla — APPROVED** [review]

**Pull request merged**